### PR TITLE
AND filter

### PR DESCRIPTION
**Pull request by SeTeM**
This change adds an ability to build the following queries:
```
county=USA && region=Idaho || country=Canada && region=Alberta
```
In additional, the PR fixes `Solr::Query::Request::OrFilter` (when filters have negations) and fixes nesting of `Solr::Query::Request::*` spec files.

### TODO
- [x] Edit README

### CODE REVIEW

File: spec/query/request/or_filter_spec.rb [hunk 1]
```diff
@@ -9,6 +9,6 @@
 
     subject { described_class.new(*filters).to_solr_s }
 
-    it { is_expected.to eq('(field:("value") OR field:(["1" TO "100"]))') }
+    it { is_expected.to eq('(-field:("value") OR field:(["1" TO "100"]))') }
```
Comment (SeTeM) [inline spec/query/request/or_filter_spec.rb]: This spec was broken.

**Comment by SeTeM** [comment]
@Valve, @drnluz, I'm going to use this code on machinio.DE tomorrow.

**Comment by Valve** [comment]
@SeTeM if possible, please add an example to the readme

**Comment by SeTeM** [comment]
@Valve, thanks for the review!
Sure, I'll add it to the readme.

**Pull request merged**